### DISCUSSION TITLE
Can't add my custom domain on Github Pages

### DISCUSSION THREAD

**Discussion by ishaanbhagat** [community/community#23051] [Other Feature Feedback, Questions, & Ideas]
<p>Hello. Today, I purchased domain name “ishaanbhagat . com” and tried to connect GithubPages to it. It worked, but the repo I used was using a Jekyll Theme that looked quite bad, and I could not find a way to fix it, so I permanently deleted my repo called “ishaanbhagat.github .io”.</p>
<p>Later the same day, I created another repo with the same name, forked from a another user’s repo. However, at the Settings &gt; Pages Menu, when I try to add “ishaanbhagat . com”, it says The CNAME <code>ishaanbhagat . com</code> is already taken. Check out <a href="https://docs.github.com/articles/troubleshooting-custom-domains#cname-errors" rel="noopener nofollow ugc">https://docs.github.com/articles/troubleshooting-custom-domains#cname-errors</a> for more information."</p>
<p>I have a feeling that the previous instance of the repository is still linked to the CNAME file with my domain, however, I am unable to access this repository as it had the same name as the new repository. The site itself shows this error code, presumably due to the linked repository being deleted.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/d/dd281fbb5216cca5202459ca3628207b0ec6349f.png" data-download-href="/uploads/short-url/vyru8IM9CymwWgFyzDdijqErwoD.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/68845756/181074576-28804910-d2f3-444a-b6fa-60f1cfef8348.png" alt="image" data-base62-sha1="vyru8IM9CymwWgFyzDdijqErwoD" width="690" height="357" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/d/dd281fbb5216cca5202459ca3628207b0ec6349f_2_690x357.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/d/dd281fbb5216cca5202459ca3628207b0ec6349f_2_1035x535.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/d/d/dd281fbb5216cca5202459ca3628207b0ec6349f_2_1380x714.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/d/dd281fbb5216cca5202459ca3628207b0ec6349f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1909×990 34.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>BTW some links have spaces in them as a new user can only upload 2 links.</p>
<p>I do realize that it probably wasn’t the best idea to permanently delete the repository I intended on working on, but I would greatly appreciate any help you can give me. Do advise me on what to do so I can connect my domain to this new site.</p>

**Comment by otter-computer** [comment]
<p>It looks like something might have gone wrong with attaching your CNAME file and it looks like it’s in a strange state where it’s only half-linked to your site :thinking: This is likely due to the filename and contents being corrupted:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/ishaanbhagat/ishaanbhagat.github.io/blob/e6cb868633d4a96b739952fe71465e3fe9d5ed09/.CNAME.un~" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/ishaanbhagat/ishaanbhagat.github.io/blob/e6cb868633d4a96b739952fe71465e3fe9d5ed09/.CNAME.un~" target="_blank" rel="noopener">ishaanbhagat/ishaanbhagat.github.io/blob/e6cb868633d4a96b739952fe71465e3fe9d5ed09/.CNAME.un~</a></h4>
<pre><code class="lang-un~">Vim�UnDo� w��B{�1Zػ�w��1���G@�&lt;I�*�`�ZE_�����`�ZD�5��</code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Can you try deleting this file and then attaching your custom domain to your site from the Pages section in your repository settings again?</p>

**Comment by ishaanbhagat** [comment]
<p>Hi Thomas,</p>
<p>Thanks for your quick response. I tried deleting the file and then trying to attach the custom domain from Pages in settings, however, I’m still getting the same error as before 🥲. Do you have any other suggestions?</p>

**Comment by jsoref** [comment]
<p>Fwiw, that’s a vim undo file:<br>
<a href="https://vimhelp.org/vim_faq.txt.html#faq-7.2" class="onebox" target="_blank" rel="noopener nofollow ugc">https://vimhelp.org/vim_faq.txt.html#faq-7.2</a></p>

**Comment by otter-computer** [comment]
<p>If you’re still seeing this error then the best thing to do is probably to delete and recreate this repository. Make sure you clone a fresh version first so you can push it back up to GitHub so you don’t lose any of your work!</p>

**Comment by ishaanbhagat** [comment]
<p>Hey Thomas,</p>
<p>I just did what you suggested, cloned a version of the repo, then deleted it and recreated with the cloned backup. However, I’m still getting the exact same error as before, saying that the domain name is already taken.</p>
<p>Is there any way for you to ‘wipe’ the domain name <code>ishaanbhagat.com</code> from Github Pages? If you go to the web address, it displays the following message</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/5/6501dde30eababd4a61dcc957760121675e26698.jpeg" data-download-href="/uploads/short-url/epydT3siira9EivsYqQ4glJmjEs.jpeg?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/68845756/181074579-ac30956a-845d-4209-b464-9d68b154cd3a.jpeg" alt="image" data-base62-sha1="epydT3siira9EivsYqQ4glJmjEs" width="517" height="282" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/5/6501dde30eababd4a61dcc957760121675e26698_2_517x282.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/6/5/6501dde30eababd4a61dcc957760121675e26698_2_775x423.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/5/6501dde30eababd4a61dcc957760121675e26698_2_1034x564.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/5/6501dde30eababd4a61dcc957760121675e26698_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2880×1572 239 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>When I had purchased the domain name, the web address was basically blank. Hence, it looks like some remnant of a deleted repo or something of that nature is still linked to the web address. Am I correct in my thinking? If so, is there anything you can do on your side to ‘de-link’ the domain name?</p>
<p>Thanks a lot for your help with this,<br>
Ishaan</p>

**Comment by otter-computer** [comment]
<p>Unfortunately not. Your site is in a stange state right now where Pages things your domain is attached but the CNAME file doesn’t exist inside your repository, which is very strange. Can you try creating this CNAME file manually and see if that helps? It should be a file called <code>CNAME</code>—no extension— with the following content:</p>
<pre><code class="lang-nohighlight">ishaanbhagat.com
</code></pre>
<p>Can you try pushing that file to your repository and seeing if it fixes this?</p>

**Comment by ishaanbhagat** [comment]
<p>Hi Thomas,</p>
<p>Thanks for your reply. Just tried creating and pushing the CNAME file, it seems to have done something but not exactly what I wanted. Now, when I try attaching my custom domain in Pages settings, I get the following message ‘No changes to custom domain.’</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/9/29849fb06fdd32d90cefa9311d5bb159c2ccbea2.jpeg" data-download-href="/uploads/short-url/5VhIJdcDjf9WnOGULPqtQ3faZF0.jpeg?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/68845756/181074582-742d5bfc-ced8-4ac5-a9b0-7c345a59c32f.jpeg" alt="image" data-base62-sha1="5VhIJdcDjf9WnOGULPqtQ3faZF0" width="690" height="372" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/9/29849fb06fdd32d90cefa9311d5bb159c2ccbea2_2_690x372.jpeg, https://aws1.discourse-cdn.com/github/optimized/3X/2/9/29849fb06fdd32d90cefa9311d5bb159c2ccbea2_2_1035x558.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/9/29849fb06fdd32d90cefa9311d5bb159c2ccbea2_2_1380x744.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/9/29849fb06fdd32d90cefa9311d5bb159c2ccbea2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2880×1554 411 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>The message stays the same no matter how many times I click it, and the actual domain still shows the ‘404 Error’ Github Pages screen.</p>
<p>Do you have any suggestions for what I could try now?</p>
<p>Thanks again for your help,<br>
Ishaan</p>

**Comment by otter-computer** [comment, marked as answer]
<p>Seems to be fixed, finally! I triggered a rebuild of your site from the backend and it looks like your domain has been attached correctly now :tada:</p>
<p>Not entirely sure what went on here. I’ll pass this on to the Pages team to investigate I think because it was especially strange, but your site should be working properly now.</p>

**Comment by ishaanbhagat** [comment]
<p>Great, thanks so much for your help Thomas!</p>

**Comment by AppInstalleriOSGH** [comment]
<p>I need help my problem is similar my GitHub account got suspended and I had a custom domain in one of the repos and now I have a new GitHub account I want to be able to use the same domain but it says it’s already in use from the suspended account. This is very urgent</p>

**Comment by otter-computer** [comment]
<p>Hi <a class="mention" href="/benjaminhornbeck6">@BenjaminHornbeck6</a>, you’ll need to <a href="https://support.github.com/request">contact the support team directly</a> to fix this as it may involve dealing with private account details.</p>
